interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeDataValidator.scala

```diff
@@ -104,8 +104,9 @@ class NodeDataValidator(modelData: ModelData, fragmentResolver: FragmentResolver
               validationContext
             )
           )
-        case a: FragmentInput => validateFragment(validationContext, outgoingEdges, compiler, a)
-        case Split(_, _) | FragmentUsageOutput(_, _, _, _) | FragmentInputDefinition(_, _, _) | BranchEndData(_) =>
+        case a: FragmentInput           => validateFragment(validationContext, outgoingEdges, compiler, a)
```

Comment:
Maybe it would be more elegant to use `fi` and `fid` instead of `a` in these two branches?

Reply:
all the other cases are just 'a' so i followed the tradition for consistency